### PR TITLE
Misc decomp 70

### PR DESCRIPTION
**Pull request by ethteck**
70%

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 70 functions (+1.14%) on `us`.

✅ This PR fixes 18 warnings!

File: include/enums.h [hunk 1]
```diff
@@ -2585,7 +2587,7 @@ enum BattleStatusFlags1 {
     BS_FLAGS1_400              = 0x00000400,
     BS_FLAGS1_800              = 0x00000800,
     BS_FLAGS1_1000             = 0x00001000,
-    BS_FLAGS1_2000             = 0x08002000,
+    BS_FLAGS1_2000             = 0x00002000,
```
Comment (nanaian) [inline include/enums.h:2590]: lol

File: src/16F740.c [hunk 2]
```diff
@@ -489,7 +564,122 @@ void btl_state_update_switch_to_player(void) {
 void btl_state_draw_switch_to_player(void) {
 }
 
-INCLUDE_ASM(s32, "16F740", btl_state_update_begin_partner_turn);
+//INCLUDE_ASM(s32, "16F740", btl_state_update_begin_partner_turn);
```
Comment (nanaian) [inline src/16F740.c:567]:
```suggestion
```

File: src/16F740.c [hunk 3]
```diff
@@ -489,7 +564,122 @@ void btl_state_update_switch_to_player(void) {
...
+                partner->koDuration--;
+                D_8029F254 = 1;
+                D_8029F258 = 20;
+                if (partner->koDuration > 0) {
+                    // TODO: We believe these are DebuffFXData*,
+                    // but unk_3C is a u8 for these and we need it to be an s32...
+                    ((s32*)(partner->debuffEffect->data))[15] = partner->koDuration;
+                } else {
+                    partner->koStatus = 0;
+                    dispatch_event_partner(0x34);
```
Comment (nanaian) [inline src/16F740.c:615]: use enum

File: src/16F740.c [hunk 4]
```diff
@@ -889,7 +1079,114 @@ void btl_state_update_end_training_battle(void) {
...
+                return;
+            }
+            script = start_script(stage->postBattle, 0xA, 0);
+            battleStatus->controlScript = script;
+            gBattleState2 = BATTLE_STATE2_UNK_3;
+            battleStatus->controlScriptID = script->id;
+            break;
+        case BATTLE_STATE2_UNK_3:
+            if (!does_script_exist(battleStatus->controlScriptID)) {
+                gBattleState2 = BATTLE_STATE2_UNK_4;
```
Comment (nanaian) [inline src/16F740.c:1133]: Worth commenting fallthrough

File: src/16F740.c [hunk 5]
```diff
@@ -926,7 +1223,162 @@ void btl_state_update_defend(void) {
...
+
+    switch (gBattleState2) {
+        case BATTLE_STATE2_UNK_0:
+            battleStatus->unk_8C = 0;
+            gBattleStatus.flags1 &= ~BS_FLAGS1_40000;
+            gBattleStatus.flags2 |= BS_FLAGS2_10 | BS_FLAGS2_8 | BS_FLAGS2_4 | BS_FLAGS2_2;
+            playerData->unk_2A6++;
+            btl_cam_use_preset(BTL_CAM_PRESET_25);
+            var_s0 = 0;
+            btl_cam_target_actor(0);
```
Comment (nanaian) [inline src/16F740.c:1246]: ACTOR_PLAYER?

File: src/battle/action_cmd/dizzy_shell.c [hunk 6]
```diff
@@ -34,30 +37,168 @@ ApiStatus func_802A9000_426170(void) {
...
+
+                if (temp_v1_2 != 0) {
+                    actionCommandStatus->barFillLevel += temp_v1_2;
+                } else {
+                    actionCommandStatus->barFillLevel += 100;
+                    if (actionCommandStatus->barFillLevel >= 500) {
+                        actionCommandStatus->barFillLevel = 500;
+                    }
+                }
+
```
Comment (nanaian) [inline src/battle/action_cmd/dizzy_shell.c:140]:
```suggestion
```

File: src/battle/action_cmd/dizzy_shell.c [hunk 7]
```diff
@@ -34,30 +37,168 @@ ApiStatus func_802A9000_426170(void) {
...
+                    if (actionCommandStatus->barFillLevel < 0) {
+                        actionCommandStatus->barFillLevel = 0;
+                    }
+                } else {
+                    actionCommandStatus->barFillLevel -= 10;
+                    if (actionCommandStatus->barFillLevel < 0) {
+                        actionCommandStatus->barFillLevel = 0;
+                    }
+                }
+
```
Comment (nanaian) [inline src/battle/action_cmd/dizzy_shell.c:124]:
```suggestion
```

File: src/battle/action_cmd/hurricane.c [hunk 8]
```diff
@@ -91,20 +95,236 @@ ApiStatus func_802A91A0_42F880(Evt* script, s32 isInitialCall) {
...
+            if (actionCommandStatus->unk_4E != 0) {
+                actionCommandStatus->unk_4E--;
+                break;
+            }
+            hud_element_set_script(actionCommandStatus->hudElements[0], &HudScript_MashAButton);
+            hud_element_set_script(actionCommandStatus->hudElements[2], &HudScript_MashBButton1);
+            actionCommandStatus->barFillLevel = 0;
+            actionCommandStatus->unk_5C = 0;
+            actionCommandStatus->unk_54 = actionCommandStatus->unk_52;
+            actionCommandStatus->state = 11;
```
Comment (nanaian) [inline src/battle/action_cmd/hurricane.c:166]: // fallthrough

File: src/battle/action_cmd/power_shock.c [hunk 9]
```diff
@@ -148,8 +147,10 @@ void func_802A9310_42D220(void) {
                     actionCommandStatus->hudElementX,
                     actionCommandStatus->hudElementY + 28);
 
-                temp_v0_2 = (((100 - actionCommandStatus->unk_46) * 60) / 100);
-                temp_v0_2 = 29 - temp_v0_2;
+                //temp_v0_2 = (((100 - actionCommandStatus->unk_46) * 60) / 100);
```
Comment (nanaian) [inline src/battle/action_cmd/power_shock.c:150]:
```suggestion
```

File: src/battle/action_cmd/water_block.c [hunk 10]
```diff
@@ -28,14 +248,372 @@ ApiStatus func_802A9398_42A888(Evt* script, s32 isInitialCall) {
...
     battleStatus->unk_86 = 0;
     actionCommandStatus->state = 10;
-    battleStatus->flags1 &= ~0x8000;
+    battleStatus->flags1 &= ~BS_FLAGS1_8000;
     func_80269118();
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "battle/action_cmd/water_block", func_802A948C_42A97C);
+//INCLUDE_ASM(s32, "battle/action_cmd/water_block", func_802A948C_42A97C);
```
Comment (nanaian) [inline src/battle/action_cmd/water_block.c:256]:
```suggestion
```

File: src/battle/action_cmd/water_block.c [hunk 11]
```diff
@@ -28,14 +248,372 @@ ApiStatus func_802A9398_42A888(Evt* script, s32 isInitialCall) {
...
+                sfx_play_sound(SOUND_234);
+            }
+            actionCommandStatus->unk_54--;
+            if (actionCommandStatus->unk_54 == 0) {
+                actionCommandStatus->unk_54 = 20;
+                actionCommandStatus->unk_5D = 0;
+                actionCommandStatus->state = 12;
+                actionCommandStatus->unk_5C = 0;
+                actionCommandStatus->unk_60 = 0;
+                return;
```
Comment (nanaian) [inline src/battle/action_cmd/water_block.c:383]: Surprised that break != return in this func

File: src/battle/area_kmr_part_3/stage/mac_01.c [hunk 12]
```diff
@@ -6,6 +6,93 @@
 ApiStatus func_80219D60_4491A0(Evt* script, s32 isInitialCall);
 INCLUDE_ASM(s32, "battle/area_kmr_part_3/448E80", func_80219D60_4491A0);
 
+// ApiStatus func_80219D60_4491A0(Evt* script, s32 isInitialCall) {
```
Comment (nanaian) [inline src/battle/area_kmr_part_3/stage/mac_01.c:9]: 🤔

**Review by nanaian — CHANGES_REQUESTED** [review]
awesome to see the battle manager stuff decompiled!!

File: src/battle/action_cmd/power_shock.c (hunk 9 above)
Comment (ethteck) [reply src/battle/action_cmd/power_shock.c:150]: It's in NON_MATCHING so this is just there to indicate what the line should really look like (though it's less nice of a score)...actually yeah I'll just remove it

File: src/battle/action_cmd/water_block.c (hunk 11 above)
Comment (ethteck) [reply src/battle/action_cmd/water_block.c:383]: I fixed a few inconsistencies here

**Pull request merged**